bundles/org.openhab.binding.senseenergy/src/main/java/org/openhab/binding/senseenergy/internal/api/SenseEnergyApi.java

```diff
@@ -362,42 +289,55 @@ private Request setHeaders(Request request) {
         return request;
     }
 
-    public void verifyToken()
-            throws InterruptedException, TimeoutException, ExecutionException, SenseEnergyApiException {
+    public void verifyToken() throws SenseEnergyApiException {
         if (tokenExpiresAt.isBefore(Instant.now())) {
             refreshToken();
         }
     }
 
-    ContentResponse sendRequest(Request request)
-            throws InterruptedException, TimeoutException, ExecutionException, SenseEnergyApiException {
+    ContentResponse sendRequest(Request request) throws SenseEnergyApiException {
         return sendRequest(request, true);
     }
 
-    ContentResponse sendRequest(Request request, boolean verifyToken)
-            throws InterruptedException, TimeoutException, ExecutionException, SenseEnergyApiException {
+    ContentResponse sendRequest(Request request, boolean verifyToken) throws SenseEnergyApiException {
         if (verifyToken) {
             verifyToken();
         }
 
         setHeaders(request);
 
-        logger.trace("REQUEST: {}", request.toString());
-        ContentResponse response = request.send();
+        ContentResponse response;
+        try {
+            logger.trace("REQUEST: {}", request.toString());
+            response = request.send();
+        } catch (Exception e) {
```

Comment:
`send` can throw multiple types of exceptions, each can lead to a different `SEVERITY`.
Atleast `IOException` and `InteruptException` should be handled different.